### PR TITLE
Implemented support for sponge economy

### PR DESCRIPTION
**Pull request by BrainStone**
Orintated on the EssentialsEco-Plugin

### CODE REVIEW

File: PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        this.economyService = economyService;
+        
+        setPluginIcon(moneyIconColored);
+    }
+
+    @Override
+    public InspectContainer getPlayerData(UUID uuid, InspectContainer inspectContainer) {
+        String name = DataCache.getInstance().getName(uuid);
+        
+        if (name == null)
```
Comment (Rsl1122) [inline PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java]: Please use brackets on one line if statements. :)

File: PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            return inspectContainer;
+        
+        Optional<UniqueAccount> uOpt = economyService.getOrCreateAccount(uuid);
+        
+        if (!uOpt.isPresent())
```
Comment (Rsl1122) [inline PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java]: Missing brackets

File: PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            BigDecimal balance = acc.getBalance(currency);
+            inspectContainer.addValue(getWithIcon(currency.getName(), moneyIconColored), currency.format(balance).toPlain());
+        }
+
+        return inspectContainer;
+    }
+
+    @Override
+    public AnalysisContainer getServerData(Collection<UUID> uuids, AnalysisContainer analysisContainer) {
+        List<UniqueAccount> players = analysisData.getValue(AnalysisKeys.PLAYERS_MUTATOR)
```
Comment (Rsl1122) [inline PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java]: This stream should use the uuids collection in the method signature :)

File: PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                .map(PlayersMutator::all).orElse(new ArrayList<>()).stream()
+                .map(container -> container.getValue(PlayerKeys.UUID)).filter(Optional::isPresent)
+                .map(Optional::get).map(economyService::getOrCreateAccount)
+                .filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList());
+        
+        for(Currency currency : economyService.getCurrencies()) {
+            BigDecimal totalBalance = BigDecimal.ZERO;
```
Comment (Rsl1122) [inline PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/sponge/SpongeData.java]: Can this block inside the for loop be refactored into a new method? :)

**Comment by Rsl1122** [comment]
Fixed merge conflict with commit 380330f

**Comment by BrainStone** [comment]
I’ll imple the requested changes soon. I’ll also rename the classes to `SpongeEconomy`

**Comment by BrainStone** [comment]
Implemented the requested changes

**Comment by Rsl1122** [comment]
Great, looks good, merged.

**Pull request merged**